apps/rush-lib/src/logic/PackageChangeAnalyzer.ts

```diff
@@ -157,6 +173,22 @@ export class PackageChangeAnalyzer {
     return projectHashDeps;
   }
 
+  private async _getIgnoreMatcherForProject(
+    project: RushConfigurationProject,
+    terminal: Terminal
+  ): Promise<Ignore> {
+    const projectConfiguration:
+      | RushProjectConfiguration
+      | undefined = await RushProjectConfiguration.tryLoadForProjectAsync(project, undefined, terminal);
```

Comment:
(Ideally this should be cached somewhere, but we don't need to solve that in this PR.)

Reply:
I'm not sure we want to do that here; this is a wrapper for the `heft-config-file` library which does is own internal caching of all config files you load with it, and it seems like that's the best way to do it.

(I did file a related ticket, https://github.com/microsoft/rushstack/issues/2691, that will improve the caching further.)